plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -613,6 +612,17 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
    void repost_exhausted_transactions(const fc::time_point& deadline);
    bool push_read_only_transaction(transaction_metadata_ptr trx, next_function<transaction_trace_ptr> next);
 
+   void set_produce_block_offset(uint32_t produce_block_offset_ms) {
+      EOS_ASSERT(produce_block_offset_ms < (config::producer_repetitions * config::block_interval_ms), plugin_config_exception,
+                 "produce-block-offset-ms ${p} must be [0 - ${max})", ("p", produce_block_offset_ms)("max", config::producer_repetitions * config::block_interval_ms));
+      _produce_block_cpu_effort_us = fc::milliseconds( config::block_interval_ms - produce_block_offset_ms / config::producer_repetitions );
+   }
+
+   fc::microseconds get_produce_block_offset() const {
```

Comment:
At the first glance, I think this returns the user supplied value.

Reply:
It does. We do not have a `milliseconds` type only a `microseconds` type.